crypto/fipsmodule/evp/p_rsa.c

```diff
@@ -635,7 +635,8 @@ static int pkey_rsa_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
   RSA *rsa = NULL;
   RSA_PKEY_CTX *rctx = ctx->data;
 
-  if (!rctx->pub_exp) {
+  // In FIPS mode, the public exponent is set within |RSA_generate_key_fips|
```

Comment:
I didn't introduce new behaviour here, and it was passing even without excluding this code in the case of FIPS build, but I thought it's better if more explicitly excluded since the exponent is not passed to `RSA_generate_key_fips`.